### DISCUSSION TITLE
Nextjs 14 always responds with 404 Not Found when using API routes

### DISCUSSION THREAD

**Discussion by stamatisk** [vercel/next.js#64447] [Help] [answered]
### Summary

I am very new to next am running the following API route code from nextjs documentation (this is a file named "hello.js" in the api folder)

export default function handler(req, res) {
  res.status(200).json({ message: 'Hello from Next.js!' })
}
but fail miserably always getting 404 not found.

Of course, if I use a Route Handlers in the App Router it works fine. Can you please tell me how I can make this API route to work?

**Comment by versecafe** [comment, marked as answer]
@stamatisk If you are doing this in the NextJS App router so under `app/` you need to use `app/path/route.ts` and then under `/path` on your site will be the API response and instead of handler you need to declare the HTTP call such as
```ts
export function GET(request: Request): Response {
  // do something
}
```

js also works if you are using it instead

```js
export function GET(request) {
  // do something
}
```

**Reply by stamatisk** [reply]
Yup I got it now, many thanks for the response

**Comment by icyJoseph** [comment]
As said before, you most likely chose to use the App Router. 

You have two paths here, change your file to `app/path/route.ts`, as said above, or move your file to `pages/api/hello.js`. 

Note that going the `app` Router way, requires you to have a named export, `GET`, rather than a default export handler.

**Reply by stamatisk** [reply]
Yup totally clear now. Many thanks!
